[1.12.2] Console spell teaching errors out for offline player

HPS Version: 1.1.9 BETA
Spigot Version: 1.12.2 r0.1

If you try to teach spells to a player through console and the player isn't online it throws an error
```
[15:24:20 WARN]: Unexpected exception while parsing console command "teach time manginoanime"
org.bukkit.command.CommandException: Unhandled exception executing 'teach time manginoanime' in com.hpspells.core.HPS$HackyCommand(teach)
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:148) ~[spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchCommand(CraftServer.java:648) ~[spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchServerCommand(CraftServer.java:634) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at net.minecraft.server.v1_12_R1.DedicatedServer.aP(DedicatedServer.java:444) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:407) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_171]
Caused by: java.lang.NullPointerException
        at com.hpspells.core.command.Teach.onCommand(Teach.java:57) ~[?:?]
        at com.hpspells.core.HPS$HackyCommand.execute(HPS.java:388) ~[?:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:141) ~[spigot-1.12.2.jar:git-Spigot-642f6d2-57ab4cf]
        ... 7 more
```
Remedies either:
- Message console saying player is not online
- Add spells to PSC for the player (If they have played before vs New player)

--- a/pom.xml
+++ b/pom.xml
@@ -3,7 +3,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>HarryPotterSpells</groupId>
     <artifactId>HarryPotterSpells</artifactId>
-    <version>1.2 BETA</version>
+    <version>1.2.1 BETA</version>
     <name>HarryPotterSpells</name>
 
     <properties>

--- a/src/com/hpspells/core/HPS.java
+++ b/src/com/hpspells/core/HPS.java
@@ -1,7 +1,6 @@
 package com.hpspells.core;
 
 import java.io.File;
-import java.io.IOException;
 import java.lang.reflect.Field;
 import java.util.ArrayList;
 import java.util.Arrays;
@@ -55,7 +54,7 @@ public class HPS extends JavaPlugin {
     public ConfigurationManager ConfigurationManager;
     public PM PM;
     public SpellManager SpellManager;
-    public Wand Wand;
+    public WandManager WandManager;
     public SpellTargeter SpellTargeter;
     public Localisation Localisation;
     public ExtensionManager ExtensionManager;
@@ -80,7 +79,7 @@ public class HPS extends JavaPlugin {
         if (localeState) {
         	SpellTargeter = new SpellTargeter(this);
             SpellManager = new SpellManager(this);
-            Wand = new Wand(this);
+            WandManager = new WandManager(this);
             ExtensionManager = new ExtensionManager(this);
             new APIHandler(this);
             
@@ -214,7 +213,7 @@ public class HPS extends JavaPlugin {
         reloadConfig();
         ConfigurationManager.reloadConfigAll();
         Localisation.load();
-        this.Wand = new Wand(this);
+        this.WandManager = new WandManager(this);
         if (!setupCrafting()) return false;
         return true;
     }
@@ -259,7 +258,7 @@ public class HPS extends JavaPlugin {
     	
         if (getConfig().getBoolean("wand.crafting.enabled", true)) {
             try {
-                ShapedRecipe wandRecipe = new ShapedRecipe(new NamespacedKey(this, "wand"), Wand.getLorelessWand());
+                ShapedRecipe wandRecipe = new ShapedRecipe(new NamespacedKey(this, "wand"), WandManager.getLorelessWand());
                 List<String> list = getConfig().getStringList("wand.crafting.recipe");
                 Set<String> ingredients = getConfig().getConfigurationSection("wand.crafting.ingredients").getKeys(false);
 

--- a/src/com/hpspells/core/Listeners.java
+++ b/src/com/hpspells/core/Listeners.java
@@ -79,7 +79,7 @@ public class Listeners implements Listener {
         }
         
         //Normal wand checking code
-        if (e.getPlayer().hasPermission(CAST_SPELLS) && HPS.Wand.isWand(e.getPlayer().getInventory().getItemInMainHand())) {
+        if (e.getPlayer().hasPermission(CAST_SPELLS) && HPS.WandManager.isWand(e.getPlayer().getInventory().getItemInMainHand())) {
             PlayerSpellConfig psc = (PlayerSpellConfig) HPS.ConfigurationManager.getConfig(ConfigurationType.PLAYER_SPELL);
 
             if (e.getPlayer().getGameMode().equals(GameMode.CREATIVE))
@@ -110,7 +110,7 @@ public class Listeners implements Listener {
                     HPS.PM.newSpell(e.getPlayer(), HPS.SpellManager.setCurrentSpellPosition(e.getPlayer(), neww).getName());
                     if (HPS.getConfig().getBoolean("wand.lore.show-current-spell")) {
                     	 //TODO: Somehow get the Wand.Lore and update it cleanly instead of this
-                        ItemStack wand = HPS.Wand.getWandFromInventory(e.getPlayer().getInventory());
+                        ItemStack wand = HPS.WandManager.getWandFromInventory(e.getPlayer().getInventory());
                         ItemMeta meta = wand.getItemMeta();
                         List<String> lore = meta.getLore();
                         for (String string : lore) {
@@ -155,7 +155,7 @@ public class Listeners implements Listener {
     @EventHandler
     public void PIEE(PlayerInteractEntityEvent e) {
         HPS.PM.debug("Fired PlayerInteractEntityEvent");
-        if (e.getPlayer().hasPermission(CAST_SPELLS) && HPS.Wand.isWand(e.getPlayer().getInventory().getItemInMainHand())) {
+        if (e.getPlayer().hasPermission(CAST_SPELLS) && HPS.WandManager.isWand(e.getPlayer().getInventory().getItemInMainHand())) {
             PlayerSpellConfig psc = (PlayerSpellConfig) HPS.ConfigurationManager.getConfig(ConfigurationType.PLAYER_SPELL);
             Integer knows = psc.getStringListOrEmpty(e.getPlayer().getName()).size() - 1, cur = HPS.SpellManager.getCurrentSpellPosition(e.getPlayer()), neww;
 
@@ -207,8 +207,8 @@ public class Listeners implements Listener {
 
     @EventHandler
     public void onPlayerCraft(CraftItemEvent e) {
-        if (HPS.Wand.isWand(e.getRecipe().getResult())) {
-            e.setCurrentItem(HPS.Wand.getWand((Player) e.getWhoClicked()));
+        if (HPS.WandManager.isWand(e.getRecipe().getResult())) {
+            e.setCurrentItem(HPS.WandManager.getWand((Player) e.getWhoClicked()));
             final Player p = (Player) e.getWhoClicked();
             Bukkit.getScheduler().runTask(HPS, new Runnable() {
 

--- a/src/com/hpspells/core/WandManager.java
+++ b/src/com/hpspells/core/WandManager.java
@@ -23,16 +23,16 @@ import com.hpspells.core.util.MiscUtilities;
 /**
  * This class manages the wand
  */
-public class Wand {
+public class WandManager {
     private HPS HPS;
     private List<Material> wandTypes = new ArrayList<Material>();
 
     /**
-     * Constructs a new {@link Wand}
+     * Constructs a new {@link WandManager}
      *
      * @param instance an instance of {@link HPS}
      */
-    public Wand(HPS instance) {
+    public WandManager(HPS instance) {
         this.HPS = instance;
         HPS.getConfig().getStringList("wand.types").forEach(string -> {
             Material material = Material.matchMaterial(string);
@@ -62,7 +62,7 @@ public class Wand {
 //        HPS.PM.broadcastMessage("No Color config: \n" + ChatColor.stripColor((String) getConfig("lore.name", "Wand")) );
 //        HPS.PM.broadcastMessage("Matching?:" + ChatColor.stripColor(itemMeta.getDisplayName()).equals(ChatColor.stripColor(getName())));
         return itemMeta.hasDisplayName() && ChatColor.stripColor(itemMeta.getDisplayName()).equals(ChatColor.stripColor(getName()));
-        //return new NBTContainerItem(i).getTag(TAG_NAME) != null;
+//        return new NBTContainerItem(i).getTag(TAG_NAME) != null;
     }
 
     public ItemStack getWand(@Nullable Player owner) {
@@ -90,7 +90,7 @@ public class Wand {
         //NBTTagCompound comp = new NBTTagCompound(TAG_NAME);
 
         if ((Boolean) getConfig("lore.enabled", true)) {
-        	Lore lore = generateLore();
+        	WandLore lore = generateLore();
         	if ((Boolean) getConfig("lore.show-current-spell", true)) {
             	Spell spell = HPS.SpellManager.getCurrentSpell(owner);
             	lore.setCurrentSpell(spell == null ? "None" : spell.getName());
@@ -198,8 +198,8 @@ public class Wand {
     	return ChatColor.translateAlternateColorCodes('&', (String) getConfig("lore.name", "Wand"));
     }
 
-    public Lore generateLore() {
-        Lore lore = new Lore();
+    public WandLore generateLore() {
+        WandLore lore = new WandLore();
         Random random = new Random();
         ConfigurationSection cs = HPS.getConfig().getConfigurationSection("wand.lore");
 
@@ -224,7 +224,7 @@ public class Wand {
         return HPS.getConfig().get("wand." + string, defaultt);
     }
 
-    public class Lore {
+    public class WandLore {
         private String currentSpell = "None", wood = "Unknown", manufacturer = "somebody", core = "Unknown", rigidity = "Unknown rigidity";
         private int woodRarity = 0, length = 13;
 

--- a/src/com/hpspells/core/api/event/wand/WandCreationEvent.java
+++ b/src/com/hpspells/core/api/event/wand/WandCreationEvent.java
@@ -1,22 +1,22 @@
 package com.hpspells.core.api.event.wand;
 
-import com.hpspells.core.Wand;
+import com.hpspells.core.WandManager;
 import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
 import org.bukkit.event.HandlerList;
 
 
 /**
- * An event called when {@link com.hpspells.core.Wand} creates a new wand
+ * An event called when {@link com.hpspells.core.WandManager} creates a new wand
  */
 public class WandCreationEvent extends Event {
     private static final HandlerList handlers = new HandlerList();
 
     private Player owner;
-    private Wand.Lore lore;
+    private WandManager.WandLore lore;
     private boolean enchantmentEffect;
 
-    public WandCreationEvent(Player owner, Wand.Lore lore, boolean enchantmentEffect) {
+    public WandCreationEvent(Player owner, WandManager.WandLore lore, boolean enchantmentEffect) {
         this.owner = owner;
         this.lore = lore;
         this.enchantmentEffect = enchantmentEffect;
@@ -54,7 +54,7 @@ public class WandCreationEvent extends Event {
      *
      * @return the lore of the wand
      */
-    public Wand.Lore getLore() {
+    public WandManager.WandLore getLore() {
         return lore;
     }
 
@@ -63,7 +63,7 @@ public class WandCreationEvent extends Event {
      *
      * @param lore the desired lore
      */
-    public void setLore(Wand.Lore lore) {
+    public void setLore(WandManager.WandLore lore) {
         this.lore = lore;
     }
 

--- a/src/com/hpspells/core/command/SpellList.java
+++ b/src/com/hpspells/core/command/SpellList.java
@@ -23,7 +23,7 @@ public class SpellList extends HCommandExecutor {
         HPS.getServer().getPluginManager().addPermission(LIST_OTHERS);
     }
 
-    public static final Permission LIST_OTHERS = new Permission("HarryPotterSpells.list.others", PermissionDefault.OP);
+    public static final Permission LIST_OTHERS = new Permission("harrypotterspells.list.others", PermissionDefault.OP);
 
     @Override
     public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {

--- a/src/com/hpspells/core/command/SpellSwitch.java
+++ b/src/com/hpspells/core/command/SpellSwitch.java
@@ -41,7 +41,7 @@ public class SpellSwitch extends HCommandExecutor {
                         HPS.PM.newSpell(player, spellName);
                         if (HPS.getConfig().getBoolean("wand.lore.show-current-spell")) {
                        	 //TODO: Somehow get the Wand.Lore and update it cleanly instead of this
-                           ItemStack wand = HPS.Wand.getWandFromInventory(player.getInventory());
+                           ItemStack wand = HPS.WandManager.getWandFromInventory(player.getInventory());
                            ItemMeta meta = wand.getItemMeta();
                            List<String> lore = meta.getLore();
                            for (String string : lore) {

--- a/src/com/hpspells/core/command/Teach.java
+++ b/src/com/hpspells/core/command/Teach.java
@@ -53,10 +53,10 @@ public class Teach extends HCommandExecutor {
             teachTo = Bukkit.getPlayer(args[1]);
         }
 
-        boolean teachOnlyKnown = (sender instanceof Player) && sender.hasPermission(teachKnown);
-        List<String> senderKnownSpells = ((PlayerSpellConfig) HPS.getConfig(ConfigurationManager.ConfigurationType.PLAYER_SPELL)).getStringListOrEmpty(teachTo.getName());
-
         if (teachTo != null) {
+            boolean teachOnlyKnown = (sender instanceof Player) && sender.hasPermission(teachKnown);
+            List<String> senderKnownSpells = ((PlayerSpellConfig) HPS.getConfig(ConfigurationManager.ConfigurationType.PLAYER_SPELL)).getStringListOrEmpty(teachTo.getName());
+            
             if (args[0].equalsIgnoreCase("all") || args[0].equalsIgnoreCase("*")) {
                 Set<Spell> spells = HPS.SpellManager.getSpells();
                 String learnedSpells = null;

--- a/src/com/hpspells/core/command/Wand.java
+++ b/src/com/hpspells/core/command/Wand.java
@@ -38,7 +38,7 @@ public class Wand extends HCommandExecutor {
                     Player player = (Player) sender;
                     if (args.length == 2) {
                         Material material = Material.matchMaterial(args[1]);
-                        if (HPS.Wand.getWandTypes().contains(material)) {
+                        if (HPS.WandManager.getWandTypes().contains(material)) {
                             giftWand(player, material);
                         } else {
                             String materialName = material == null ? "NULL" : material.toString();
@@ -57,7 +57,7 @@ public class Wand extends HCommandExecutor {
                 if (target != null) {
                     if (args.length == 2) {
                         Material material = Material.matchMaterial(args[1]);
-                        if (HPS.Wand.getWandTypes().contains(material)) {
+                        if (HPS.WandManager.getWandTypes().contains(material)) {
                             giftWand(target, material);
                         } else {
                             String materialName = material == null ? "NULL" : material.toString();
@@ -86,7 +86,7 @@ public class Wand extends HCommandExecutor {
      * @param material The material to use if required
      */
     public void giftWand(Player player, Material material) {
-        player.getInventory().setItem(player.getInventory().firstEmpty(), HPS.Wand.getWand(player, material));
+        player.getInventory().setItem(player.getInventory().firstEmpty(), HPS.WandManager.getWand(player, material));
         HPS.PM.tell(player, HPS.Localisation.getTranslation("cmdWanGiven"));
         if (HPS.getConfig().getBoolean("wand.explosion-effect")) {
             player.getWorld().createExplosion(player.getLocation(), 0, false);

--- a/src/com/hpspells/core/spell/Arrow.java
+++ b/src/com/hpspells/core/spell/Arrow.java
@@ -1,0 +1,27 @@
+package com.hpspells.core.spell;
+
+import org.bukkit.entity.Player;
+
+import com.hpspells.core.HPS;
+import com.hpspells.core.spell.Spell.SpellInfo;
+
+@SpellInfo(
+        name = "Arrow",
+        description = "descArrow",
+        goThroughWalls = false,
+        cooldown = 90
+)
+public class Arrow extends Spell {
+
+    public Arrow(HPS instance) {
+        super(instance);
+    }
+
+    @Override
+    public boolean cast(Player p) {
+        Double velocityModifier = (Double) getConfig("velocity", 1.2);
+        p.launchProjectile(org.bukkit.entity.Arrow.class, p.getLocation().getDirection().multiply(velocityModifier));
+        return true;
+    }
+
+}

--- a/src/com/hpspells/core/spell/Duro.java
+++ b/src/com/hpspells/core/spell/Duro.java
@@ -1,0 +1,48 @@
+package com.hpspells.core.spell;
+
+import org.bukkit.Color;
+import org.bukkit.Material;
+import org.bukkit.Particle;
+import org.bukkit.Particle.DustOptions;
+import org.bukkit.block.Block;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+
+import com.hpspells.core.HPS;
+import com.hpspells.core.SpellTargeter.SpellHitEvent;
+import com.hpspells.core.spell.Spell.SpellInfo;
+import com.hpspells.core.util.HPSParticle;
+
+@SpellInfo(
+        name = "Duro",
+        description = "descDuro",
+        range = 50,
+        goThroughWalls = false,
+        cooldown = 90,
+        icon = Material.STONE
+)
+public class Duro extends Spell {
+
+    public Duro(HPS instance) {
+        super(instance);
+    }
+
+    @Override
+    public boolean cast(Player p) {
+        HPS.SpellTargeter.register(p, new SpellHitEvent() {
+            
+            @Override
+            public void hitEntity(LivingEntity entity) {
+                hitBlock(entity.getEyeLocation().getBlock());
+            }
+
+            @Override
+            public void hitBlock(Block block) {
+                block.setType(Material.STONE);
+            }
+
+        }, 1, 0.5, 2, new HPSParticle(Particle.REDSTONE, new DustOptions(Color.GRAY, 1)));
+        return false;
+    }
+
+}

--- a/src/com/hpspells/core/spell/Expelliarmus.java
+++ b/src/com/hpspells/core/spell/Expelliarmus.java
@@ -11,6 +11,7 @@ import org.bukkit.block.Block;
 import org.bukkit.entity.Item;
 import org.bukkit.entity.LivingEntity;
 import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
 import org.bukkit.util.Vector;
 
 import com.hpspells.core.HPS;
@@ -43,11 +44,13 @@ public class Expelliarmus extends Spell {
             @Override
             public void hitEntity(LivingEntity entity) {
                 Location targetloc = entity.getLocation();
-                List<Material> disarmItems = new ArrayList<Material>(Arrays.asList(Material.matchMaterial(HPS.getConfig().getString("wand.type"))));
+                //HPS.getConfig().getStringList("wand.types").stream().map(Material::matchMaterial).collect(Collectors.toList())
+                List<Material> disarmItems = new ArrayList<Material>();
                 if ((Boolean) getConfig("disarm-weapons", true))
-                    disarmItems.addAll(Arrays.asList(Material.STICK, Material.WOODEN_SWORD, Material.STONE_SWORD, Material.IRON_SWORD, Material.GOLDEN_SWORD, Material.DIAMOND_SWORD, Material.BOW));
+                    disarmItems.addAll(Arrays.asList(Material.WOODEN_SWORD, Material.STONE_SWORD, Material.IRON_SWORD, Material.GOLDEN_SWORD, Material.DIAMOND_SWORD, Material.BOW));
 
-                if (disarmItems.contains(entity.getEquipment().getItemInMainHand().getType())) {
+                ItemStack itemStack = entity.getEquipment().getItemInMainHand();
+                if (disarmItems.contains(itemStack.getType()) || HPS.WandManager.isWand(itemStack)) {
                     Item i = entity.getWorld().dropItem(targetloc, entity.getEquipment().getItemInMainHand());
                     entity.getEquipment().setItemInMainHand(null);
                     Vector vector = targetloc.getDirection();

--- a/src/com/hpspells/core/spell/SpellManager.java
+++ b/src/com/hpspells/core/spell/SpellManager.java
@@ -1,13 +1,16 @@
 package com.hpspells.core.spell;
 
-import com.google.common.collect.Iterables;
-import com.hpspells.core.HPS;
-import com.hpspells.core.api.event.SpellPostCastEvent;
-import com.hpspells.core.api.event.SpellPreCastEvent;
-import com.hpspells.core.configuration.ConfigurationManager.ConfigurationType;
-import com.hpspells.core.configuration.PlayerSpellConfig;
-import com.hpspells.core.spell.Spell.SpellInfo;
-import com.hpspells.core.util.ReflectionsReplacement;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.logging.Level;
+
+import javax.annotation.Nullable;
+
 import org.bukkit.Bukkit;
 import org.bukkit.Effect;
 import org.bukkit.Location;
@@ -18,9 +21,15 @@ import org.bukkit.event.Listener;
 import org.bukkit.permissions.Permission;
 import org.bukkit.permissions.PermissionDefault;
 
-import javax.annotation.Nullable;
-import java.util.*;
-import java.util.logging.Level;
+import com.google.common.collect.Iterables;
+import com.hpspells.core.HPS;
+import com.hpspells.core.api.event.SpellEvent;
+import com.hpspells.core.api.event.SpellPostCastEvent;
+import com.hpspells.core.api.event.SpellPreCastEvent;
+import com.hpspells.core.configuration.ConfigurationManager.ConfigurationType;
+import com.hpspells.core.configuration.PlayerSpellConfig;
+import com.hpspells.core.spell.Spell.SpellInfo;
+import com.hpspells.core.util.ReflectionsReplacement;
 
 /**
  * A class that manages spells and holds lots of spell related utilities
@@ -191,13 +200,15 @@ public class SpellManager {
     }
 
     /**
-     * Casts a spell cleverly; checking permissions, sending effects ect
+     * Casts a spell cleverly. Checks permissions, triggering {@link SpellEvent}'s , sending effects and applying cooldown.
+     * 
+     * Note: Does not check if player has a wand in their inventory
      *
      * @param player the player who is casting
      * @param spell  the spell that they are casting
      */
     public void cleverCast(Player player, Spell spell) {
-        if (!player.hasPermission("harrypotterspells.cast") || !spell.playerKnows(player) || !player.getInventory().contains(Material.STICK))
+        if (!player.hasPermission("harrypotterspells.cast") || !spell.playerKnows(player))
             return;
 
         PlayerSpellConfig psc = (PlayerSpellConfig) HPS.ConfigurationManager.getConfig(ConfigurationType.PLAYER_SPELL);

--- a/src/cooldown.yml
+++ b/src/cooldown.yml
@@ -21,6 +21,7 @@ cooldowns:
   confundo: 180
   crucio: 300
   deprimo: 180
+  duro: 90
   enderchest: -1
   episkey: 60
   evanesco: 45

--- a/src/cooldown.yml
+++ b/src/cooldown.yml
@@ -12,6 +12,7 @@ cooldowns:
   alarte-ascendare: 45
   aparecium: 300
   arania-exumai: 90
+  arrow: 90
   avada-kedavra: 300
   avis: 300
   bauleo: 10

--- a/src/plugin.yml
+++ b/src/plugin.yml
@@ -1,5 +1,5 @@
 name: HarryPotterSpells
 main: com.hpspells.core.HPS
 prefix: "HPS"
-version: 1.2 BETA
+version: 1.2.1 BETA
 api-version: 1.13

--- a/src/spells.yml
+++ b/src/spells.yml
@@ -19,6 +19,9 @@ spells:
   arania-exumai:
     # How much the spider is knocked backwards
     knockback: 10
+  arrow:
+    # The velocity of the arrow that gets shot out of the wand
+    velocity: 1.2
   avis:
     # How randomly should the bats or chickens be spread out
     randomness: 1

--- a/src/us-english.properties
+++ b/src/us-english.properties
@@ -140,6 +140,7 @@ descConfringo = Launches a fireball from the tip of your wand
 descConfundo = Confuses your target
 descCrucio = Torture your victim
 descDeprimo = Slows your target to an almost halt
+descDuro = Turns a block into stone
 descEnderChest = Opens your personal enderchest
 descEpiskey = Slowly heals your target
 descEvanesco = Makes the caster invisible

--- a/src/us-english.properties
+++ b/src/us-english.properties
@@ -131,6 +131,7 @@ descAlarteAscendare = Propels the targeted mob upward
 descAlohomora = Unlocks the targeted iron door
 descAparecium = Makes nearby players visible
 descAraniaExumai = Damages and knocks back spiders
+descArrow = Shoots an arrow out of the wand
 descAvadaKedavra = Kills whomever is targeted
 descAvis = Shoots a flock of chickens from your wand
 descBauleo = Causes all nearby dropped items to be stored into a nearby chest

--- a/src/us-english.properties
+++ b/src/us-english.properties
@@ -156,6 +156,7 @@ descMulticorfors = Changes the colour of any targeted wool or sheep
 descObscuro = Blinds the target mob or player
 descOrchideous = Plants a red rose on the target block
 descPetrificusTotalus = Stuns the target player
+descPointMe = Sets the player's view to the north
 descReducto = Creates an explosion
 descRefillingCharm = Refills bowls with mushroom soup, buckets with water or milk (defined in config)
 descReparo = Repairs the item in your hand fully

